default.nix

```diff
@@ -0,0 +1,7 @@
+{system ? builtins.currentSystem}: let
+  # Import nixpkgs and dream2nix
+  helpers = import ./helpers.nix {inherit system;};
+  inherit (helpers) importNgiPackages pkgs;
+  ngiPackages = importNgiPackages pkgs;
```

Comment:
Defining inputs as arguments may be a slightly healthier pattern. This is a lot of speculation on my side though, because there's nothing established that I've seen, but all other patterns I've seen seem not as healthy. Here's [an example of what that could look like](https://github.com/fricklerhandwerk/lazy-drv/blob/6260dc5c0a542fe969174613df4b8f2352176002/default.nix#L18-L27).

One cool thing about the Nix language is that destructured function arguments have their own scope, and you can do arbitrary computation in default values.